bin/hdfcoinc/pycbc_fit_sngls_split_binned

```diff
@@ -183,16 +195,68 @@ boundaries = trigf[args.ifo + '/template_boundaries'][:]
 max_boundary_id = np.argmax(boundaries)
 sorted_boundary_list = np.sort(boundaries)
 
-logging.info('getting stat values from trigger file')
+logging.info('Processing template boundaries')
+where_idx_end = np.zeros_like(boundaries)
+for idx, idx_start in enumerate(boundaries):
+    if idx == max_boundary_id:
+        where_idx_end[idx] = trigf[args.ifo+'/end_time'].size
+    else:
+        where_idx_end[idx] = sorted_boundary_list[
+            np.argmax(sorted_boundary_list == idx_start) + 1]
+
+logging.info('calculating single stat values from trigger file')
 stat = get_stat(args.sngl_stat, trigf[args.ifo])
 
 logging.info('applying DQ vetoes')
 time = trigf[args.ifo+'/end_time'][:]
 remove, junk = events.veto.indices_within_segments(time, [args.veto_file],
                      ifo=args.ifo, segment_name=args.veto_segment_name)
 stat[remove] = np.zeros_like(remove)
+time[remove] = np.zeros_like(remove)
 logging.info('{} out of {} trigs removed after vetoing with {} from {}'.format(
                   remove.size, stat.size, args.veto_segment_name, args.veto_file))
+
+logging.info('Applying pruning around loudest triggers in each split up to a maximum of {}'.format(
+                  args.prune_number))
+for x in range(args.split_one_nbins):
+    if not args.prune_number: break
+    id_bin1 = id_in_bin1[x]
+    for y in range(args.split_two_nbins):
+        id_bin2 = id_in_bin2[y]
+        id_in_both = np.intersect1d(id_bin1, id_bin2)
+        if len(id_in_both) == 0: continue
+        vals_inbin = []
+        time_inbin = []
+        for idx in id_in_both:
+            where_idx_start = boundaries[idx]
+            vals_inbin += list(stat[where_idx_start:where_idx_end[idx]])
+            time_inbin += list(time[where_idx_start:where_idx_end[idx]])
+
+        vals_inbin = np.array(vals_inbin)
+        time_inbin = np.array(time_inbin)
+
+        count_pruned = 0
+        logging.info('Pruning in split {}-{} {}-{}'.format(args.split_param_one, x,
+                                     args.split_param_two, y))
+        while count_pruned < args.prune_number:
+            max_val_arg = vals_inbin.argmax()
+            max_statval = vals_inbin[max_val_arg]
+
+            logging.info('Prune {}: pruning triggers around time {}s'.format(
+                                     count_pruned, args.prune_window,
+                                     time[max_val_arg]))
+            remove = np.nonzero(abs(time_inbin[max_val_arg] - time)
+                                    < args.prune_window)[0]
+            remove_inbin = np.nonzero(abs(time_inbin[max_val_arg] - time_inbin)
+                                      < args.prune_window)[0]
+            logging.info('Removing {} triggers, {} in this split'
```

Comment:
seems a bit over-verbose overall. can you combine this log message with the previous one? & for instance you don't have to tell the user the size of the prune window since it's already in the command line